kedro/extras/datasets/pandas/csv_dataset.py

```diff
@@ -137,6 +137,7 @@ def __init__(
 
         _fs_open_args_load.setdefault("mode", "r")
         _fs_open_args_save.setdefault("mode", "w")
+        _fs_open_args_save.setdefault("newline", "")
```

Comment:
I mean changing `_fs_open_args_load` to set the mode on the `fsspec` open :)

Reply:
sorry, I don't understand, what is your proposed changes?

Reply:
Is the problem fixed if you set the default fsspec load arg to be `mode: “rb”` instead of `mode: “r”`?

Reply:
the problem is the output csv, it should be related to save args only.

Reply:
sorry - then would making save args default to "wb" work?

Reply:
I think I replied this earlier? As I said wb is not possible with pandas to_csv method

Reply:
import pandas as pd
df = pd.DataFrame([1,2,])
df
0
0 1
1 2
with open('file.csv', 'wb') as f:
... df.to_csv(f)



this should give u an error.

Reply:
i see, i see! my bad :)

Reply:
So is this fix ok?

Supplementing the doc of python csv module suggest to always use newline=''  (scroll to the bottom footnote section)

https://docs.python.org/3/library/csv.html#csv.writer